jerry-core/CMakeLists.txt

```diff
@@ -17,23 +17,24 @@ set(JERRY_CORE_NAME jerry-core)
 project (${JERRY_CORE_NAME} C)
 
 # Optional features
-set(FEATURE_CPOINTER_32_BIT  OFF     CACHE BOOL   "Enable 32 bit compressed pointers?")
-set(FEATURE_DEBUGGER         OFF     CACHE BOOL   "Enable JerryScript debugger?")
-set(FEATURE_ERROR_MESSAGES   OFF     CACHE BOOL   "Enable error messages?")
-set(FEATURE_EXTERNAL_CONTEXT OFF     CACHE BOOL   "Enable external context?")
-set(FEATURE_JS_PARSER        ON      CACHE BOOL   "Enable js-parser?")
-set(FEATURE_MEM_STATS        OFF     CACHE BOOL   "Enable memory statistics?")
-set(FEATURE_MEM_STRESS_TEST  OFF     CACHE BOOL   "Enable mem-stress test?")
-set(FEATURE_PARSER_DUMP      OFF     CACHE BOOL   "Enable parser byte-code dumps?")
-set(FEATURE_PROFILE          "es5.1" CACHE STRING "Use default or other profile?")
-set(FEATURE_REGEXP_DUMP      OFF     CACHE BOOL   "Enable regexp byte-code dumps?")
-set(FEATURE_SNAPSHOT_EXEC    OFF     CACHE BOOL   "Enable executing snapshot files?")
-set(FEATURE_SNAPSHOT_SAVE    OFF     CACHE BOOL   "Enable saving snapshot files?")
-set(FEATURE_SYSTEM_ALLOCATOR OFF     CACHE BOOL   "Enable system allocator?")
-set(FEATURE_VALGRIND         OFF     CACHE BOOL   "Enable Valgrind support?")
-set(FEATURE_VALGRIND_FREYA   OFF     CACHE BOOL   "Enable Valgrind-Freya support?")
-set(FEATURE_VM_EXEC_STOP     OFF     CACHE BOOL   "Enable VM execution stopping?")
-set(MEM_HEAP_SIZE_KB         "512"   CACHE STRING "Size of memory heap, in kilobytes")
+set(FEATURE_CPOINTER_32_BIT    OFF     CACHE BOOL   "Enable 32 bit compressed pointers?")
+set(FEATURE_DEBUGGER           OFF     CACHE BOOL   "Enable JerryScript debugger?")
+set(FEATURE_ERROR_MESSAGES     OFF     CACHE BOOL   "Enable error messages?")
+set(FEATURE_EXTERNAL_CONTEXT   OFF     CACHE BOOL   "Enable external context?")
+set(FEATURE_JS_PARSER          ON      CACHE BOOL   "Enable js-parser?")
+set(FEATURE_MEM_STATS          OFF     CACHE BOOL   "Enable memory statistics?")
+set(FEATURE_MEM_STRESS_TEST    OFF     CACHE BOOL   "Enable mem-stress test?")
+set(FEATURE_PARSER_DUMP        OFF     CACHE BOOL   "Enable parser byte-code dumps?")
+set(FEATURE_PROFILE            "es5.1" CACHE STRING "Use default or other profile?")
+set(FEATURE_REGEXP_STRICT_MODE OFF     CACHE BOOL   "Enable regexp strict mode?")
```

Comment:
Do we change the current behaviour here? Personally I would prefer FEATURE_REGEXP_COMPATIBILITY_MODE, and keep it OFF by default.

Reply:
Yes, now the compatibility mode is the default, so the regexp engine accepts this `/{}/` regex pattern while the strict mode does not.
Shall I set the strict mode as default?
(Personally I would prefer the extended behaviour as default, like other engines.)

Reply:
I would like to hear others opinion (kind of voting). I usually prefer to keep the current behaviour but I am not that strict.

Reply:
I don't have strong preferences. I think the average user won't notice the compatibility features only if they dig into the build system. So enable compatibility mode by default sounds reasonable for me.

Reply:
I agree with @szledan and @LaszloLango, the compatibility features should be enabled by default, but I would change the name also. So my vote is to set the `FEATURE_REGEXP_COMPATIBILITY_MODE` to `ON` instead of setting the `FEATURE_REGEXP_STRICT_MODE` to `OFF`.

Reply:
Sorry, I rethought it, the `FEATURE_REGEXP_STRICT_MODE` with `OFF` is fine. The other version may cause problems/different behaviors if somebody want to compile without CMake. LGTM (informally)